### PR TITLE
fix(testing): Create root babel config if one isn't present

### LINKED ISSUE #6782: @nrwl/node library babel jest setting does not work
## Current Behavior
After scaffolding a brand new node library using @nrwl/node and selecting the `babelJest` option tests do not run for the generated library.

## Expected Behavior
Tests should be able to run in a new project

## Steps to Reproduce
- Generate a new NX workspace
- Add the `@nrwl/node` package
- Generate a node library choosing the `babelJest` option, eg:
> npm run nx generate @nrwl/node:library -- --name=babel-test \
  --buildable \
  --importPath=@some-org/babel-test \
  --publishable \
  --standaloneConfig \
  --strict \
  --no-interactive \
  --babelJest
- Run tests for the newly create library `npm run test babel-test:test`
- Observe that the tests fail to run

### Failure Logs
    FAIL   babel-test  packages/babel-test/src/lib/babel-test.spec.ts
    ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /home/jesse/code/deepkit-community/modules/packages/babel-test/src/lib/babel-test.spec.ts:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import { babelTest } from './babel-test';
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (../../node_modules/jest-runtime/build/index.js:1479:14)
      at TestScheduler.scheduleTests (../../node_modules/@jest/core/build/TestScheduler.js:333:13)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.218 s

### Environment
```
Node : 14.15.4
  OS   : linux x64
  npm  : 6.14.14
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.7.2
  @nrwl/cypress : Not Found
  @nrwl/devkit : 12.7.2
  @nrwl/eslint-plugin-nx : 12.7.2
  @nrwl/express : Not Found
  @nrwl/jest : 12.7.2
  @nrwl/linter : 12.7.2
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : 12.7.2
  @nrwl/nx-cloud : 12.3.10
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.7.2
  @nrwl/web : Not Found
  @nrwl/workspace : 12.7.2
  @nrwl/storybook : Not Found
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

### PR DESCRIPTION
**Pull request by barbados-clemens**
## Expected Behavior

add changes to create a babel config when one isn't present and babel-jest is selected. 
add `@nrwl/web` as dev dep when babel-jest is selected to have babel deps needed by babel-jest

## Related Issue(s)

Fixes #6782

TODO: 
- [x] Migration
- [x] Testing
- [x] Validation

### CODE REVIEW

File: packages/jest/src/generators/jest-project/lib/create-files.ts [hunk 1]
```diff
@@ -24,4 +25,13 @@ export function createFiles(tree: Tree, options: JestProjectSchema) {
   if (options.setupFile === 'none') {
     tree.delete(join(projectConfig.root, './src/test-setup.ts'));
   }
+
+  if (options.babelJest && !tree.exists('babel.config.json')) {
```
Comment (Cammisuli) [inline packages/jest/src/generators/jest-project/lib/create-files.ts:29]: What happens if the root babel config is `.babelrc`?

File: packages/jest/src/generators/init/init.ts [hunk 2]
```diff
@@ -68,6 +68,21 @@ function updateDependencies(tree: Tree, options: NormalizedSchema) {
...
+    devDeps['@babel/plugin-proposal-decorators'] = '^7.14.5';
+    devDeps['@babel/plugin-transform-regenerator'] = '^7.14.5';
+    devDeps['@babel/plugin-transform-runtime'] = '^7.15.0';
+    devDeps['@babel/runtime'] = '^7.14.8';
+    devDeps['@babel/preset-typescript'] = '^7.15.0';
+    devDeps['babel-loader'] = '^8.2.2';
+    devDeps['babel-plugin-const-enum'] = '^1.0.1';
+    devDeps['babel-plugin-macros'] = '^2.8.0';
+    devDeps['babel-plugin-transform-async-to-promises'] = '^0.8.15';
+    devDeps['babel-plugin-transform-typescript-metadata'] = '^0.3.1';
```
Comment (Cammisuli) [inline packages/jest/src/generators/init/init.ts:85]: These are a lot 😦 

And I agree with your comment, this should be a function somewhere. If not, then at least these version numbers should be in a constant file; similar to `babelJestVersion`

File: packages/jest/src/generators/jest-project/lib/create-files.ts (hunk 1 above)
Comment (Cammisuli) [reply packages/jest/src/generators/jest-project/lib/create-files.ts:29]: Apparently we don't generate a root workspace babelrc file 🤷🏼

File: packages/jest/src/generators/init/init.ts (hunk 2 above)
Comment (barbados-clemens) [reply packages/jest/src/generators/init/init.ts]: moved them out into a const 👍

File: packages/web/babel.ts [hunk 3]
```diff
@@ -16,6 +17,9 @@ interface NxReactBabelPresetOptions {
 }
 
 module.exports = function (api: any, options: NxReactBabelPresetOptions = {}) {
+  console.warn(
```
Comment (nartc) [inline packages/web/babel.ts:20]: suggestion: maybe use `logger` from `nrwl/devkit` (which uses `chalk` under the hood) for better warning mesage

**Review by nartc — COMMENTED** [review]
One small suggestion. 

question: we still need the migrations in this PR or is it going to be a separate one?

**Comment by barbados-clemens** [comment]
> One small suggestion.
> 
> question: we still need the migrations in this PR or is it going to be a separate one?

No, I'll write the migration for this change in this PR.

**Review by nartc — APPROVED** [review]

**Pull request merged**